src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,167 @@
+import { HTMLAttributes, ReactNode, useId } from "react";
+import styled from "styled-components";
+import { IconButton, Label } from "@/components";
+import { Error, FormElementContainer, FormRoot } from "../commonElement";
+
+interface CommonProgressBarProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, "children"> {
+  progress: number;
+  label?: ReactNode;
+  error?: ReactNode;
+  orientation?: "vertical" | "horizontal";
+  dir?: "start" | "end";
+}
+
+interface DefaultProgressBar extends CommonProgressBarProps {
+  type?: "default";
+  successMessage?: ReactNode;
+}
+
+interface SmallProgressBar extends CommonProgressBarProps {
+  type: "small";
+  successMessage?: never;
+  dismissable?: never;
+  onCancel?: never;
+}
+interface DismissableProgressBar {
+  dismissable: true;
+  onCancel: () => void;
+}
+interface NonDismissableProgressBar {
+  dismissable?: false;
+  onCancel?: never;
+}
+export type ProgressBarProps =
+  | (DefaultProgressBar & (DismissableProgressBar | NonDismissableProgressBar))
+  | SmallProgressBar;
+const ProgressContainer = styled.div<{
+  $progress: number;
+  $completed: boolean;
+  $type: "small" | "default";
+  $error: boolean;
+}>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  overflow: hidden;
+  transition: all 100ms ease-in-out;
+  width: 100%;
+  width: -webkit-fill-available;
+  width: fill-available;
+  width: stretch;
+  min-height: 2px;
+  ${({ theme, $completed, $progress, $type, $error }) => `
+    background: ${
+      $completed
+        ? theme.click.field.color.background.default
+        : `linear-gradient(to right, ${theme.global.color.accent.default} 0%, ${theme.global.color.accent.default} ${$progress}%, ${theme.click.field.color.background.default} ${$progress}%,  ${theme.click.field.color.background.default} 100%)`
+    };
+    background-size: calc(100% + 2px);
+    background-position: center;
+    gap: ${theme.click.field.space.gap};
+    border-radius: ${theme.click.field.radii.all};
+    font: ${theme.typography.styles.product.text.normal.sm};
+    padding: ${$type === "default" ? "0.25rem" : 0} ${theme.click.field.space.x};
+    padding-right: 0;
+    color: ${theme.global.color.accent.default};
+    border: 1px solid ${theme.click.field.color.stroke.default};
```

Comment:
I guess this is because we don't have proper tokens right?

If so, I would leave a todo pointing to the issue.

Reply:
Added the todo on top of the styled component